### PR TITLE
remove closures in driver

### PR DESCRIPTION
**Pull request by juliasloan25**
## Purpose 
Remove closures over global variables in `coupler_driver.jl` to improve performance.

This includes changes for:
- comms_ctx
- mono_surface
- turbulent_fluxes
- thermo_params

### CODE REVIEW

File: config/model_configs/interactive_debug.yml [hunk 1]
```diff
@@ -17,6 +17,6 @@ rad: "gray"
 run_name: "interactive_debug_run"
 start_date: "19790301"
 surface_setup: "PrescribedSurface"
-t_end: "20days"
+t_end: "200secs"
```
Comment (LenkaNovak) [inline config/model_configs/interactive_debug.yml:20]: can we change this back to 20days when it's ready?

File: experiments/AMIP/coupler_driver.jl [hunk 2]
```diff
@@ -573,32 +589,22 @@ depend on initial conditions of other component models than those in which the v
...
 
-# 4.surface vapor specific humidity (`q_sfc`): step surface models with the new surface density to calculate their respective `q_sfc` internally
+# 3.surface vapor specific humidity (`q_sfc`): step surface models with the new surface density to calculate their respective `q_sfc` internally
 ## TODO: the q_sfc calculation follows the design of the bucket q_sfc, but it would be neater to abstract this from step! (#331)
 step!(land_sim, Δt_cpl)
 step!(ocean_sim, Δt_cpl)
 step!(ice_sim, Δt_cpl)
 
-# 5.turbulent fluxes: now we have all information needed for calculating the initial turbulent surface fluxes using the combined state
+# 4.turbulent fluxes: now we have all information needed for calculating the initial turbulent surface fluxes using the combined state
```
Comment (LenkaNovak) [inline experiments/AMIP/coupler_driver.jl:607]: `turbulent_fluxes` and `thermo_params` can also be called from cs in this section.

**Review by LenkaNovak — COMMENTED** [review]
Thanks for doing this, @juliasloan25. So far, only a couple of minor comments and happy to approve once CI passes.

File: config/model_configs/interactive_debug.yml (hunk 1 above)
Comment (juliasloan25) [reply config/model_configs/interactive_debug.yml:20]: oops yes, I didn't mean to include this

File: src/Interfacer.jl [hunk 3]
```diff
@@ -65,6 +67,8 @@ struct CoupledSimulation{
     diagnostics::TD
     callbacks::NTC
     dirs::NTP
+    turbulent_fluxes::TF
+    thermo_params::TP
```
Comment (charleskawczynski) [inline src/Interfacer.jl:71]: We could make this a keyword struct, if that simplifies how it's called. Either way is fine, though.

**Review by charleskawczynski — APPROVED** [review]

File: src/Interfacer.jl (hunk 3 above)
Comment (juliasloan25) [reply src/Interfacer.jl:71]: I think I'll just keep it as it is, but I'll keep this in mind!

**Pull request merged**